src/validate.ts

```diff
@@ -1,113 +1,33 @@
-  customData?: {[key: string]: any};
-}
-
-interface QueryInput {
-  intent: string; 
-  payload: {
-    devices: Device[]; 
-  };
-}
-
-interface QueryRequest {
-  requestId: string;
-  inputs: QueryInput[];
-}
-
-/**
- * Generates QUERY request 
- * @param devices Array of type Device containing device IDs and respective custom data
- * @returns Specified format for QUERY intent request. 
- */
-export function generateQueryRequest(devices: Device[]) : QueryRequest{
-    const requestId = generateRequestID(100,999);
-    return {
-        requestId,
-        inputs: [{
-          intent: "action.devices.QUERY",
-          payload: {
-            devices
-            }
-        }]
-    }
-}
-
-
-interface Command{
-    name: string,
-    params?: {[key: string]: any}
-}
-
-interface ExecuteInput{
-    intent: string;
-    payload: {
-        devices: Device[];
-        execution: Command[];
-    };
-}
-
-interface ExecuteRequest{
-    requestId: string;
-    inputs: ExecuteInput[];
-}
+const ajv = new Ajv({
+  allErrors: true,
+});
 
 /**
- * Generates EXECUTE request 
- * @param devices Array of type Device containing device IDs and respective custom data 
- * @param execution Array of type Command specifying the commands of the devices and their respective parameters 
- * @returns Specified format for EXECUTE intent request. 
+ * Helper function that uses AJV library to validate the response against the schema 
+ * @param apiResponse User defined api response
+ * @return Returns undefined if valid is true, returns an array of error(s) if valid is false. 
  */
-export function generateExecuteRequest(devices: Device[], execution: Command[]): ExecuteRequest{
-    const requestId = generateRequestID(100,999);
-    return {
-        requestId,
-        inputs: [{
-          intent: "action.devices.EXECUTE",
-          payload: {
-            devices,
-            execution
-          }
-        }]
-    }
+function validateSyncResponse(apiResponse: object) {
+  const syncResponseSchema = require('../intents/sync.response.schema.json');
+  const isValid = ajv.validate(syncResponseSchema, apiResponse);
+  if (isValid) {
+    return undefined;
+  }
+  return ajv.errors;
 }
 
 /**
- * Generates DISCONNECT request 
- * @returns Specified format for DISCONNECT intent request. 
+ * Identifies the response type and validates the function based on the schemas.
+ * @param apiResponse User defined api response 
+ * @param responseType User defined intent response 
+ * @return Errors from AJV validation, if any. Undefined otherwise. 
  */
-export function generateDisconnectRequest(){
-    const requestId = generateRequestID(100,999);
-    return {
-        requestId,
-        inputs: [{
-          intent: "action.devices.DISCONNECT"
-        }]
-    }
+export function validate(apiResponse: object, responseType: 'sync' | 'query'|'execute'| 'disconnect') {
+  if (responseType == 'sync') {
+    return validateSyncResponse(apiResponse);
+  }
+  return undefined;
```

Comment:
Would there be an error here rather than `undefined`, as an invalid `responseType` was selected.

Reply:
Yes that's true. Since the types of responses are specified in the function, it'll throw an error automatically without specifying. But, maybe instead of returning undefined a better way would be to throw new Error(); instead? Lmk your thoughts